Exiled.API/Extensions/RoomExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="RoomExtensions.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Extensions
+{
+    using System.Collections.Generic;
+    using Exiled.API.Enums;
+
+    /// <summary>
+    /// A set of extensions for <see cref="RoomType"/>.
+    /// </summary>
+    public static class RoomExtensions
+    {
+        /// <summary>
+        /// Checks if a <see cref="RoomType">room type</see> is a gate.
+        /// </summary>
+        /// <param name="room">The room to be checked.</param>
+        /// <returns>Returns whether the <see cref="RoomType"/> is a gate or not.</returns>
+        public static bool IsGate(this RoomType room) => room is RoomType.EzGateA or RoomType.EzGateB;
```

Comment:
I would put an other method like this HasGate for all the room who have a Gate

Reply:
I don't really agree, in fact, many rooms can and should `true` because they just have a `ContainmentDoor` inside them. For instance, let's take the Gate A and the room after the main Gate A's room, following your logic/idea, we should check the next room whether it has a gate, and we'd expect it to return `true`.
Hence, I prefer `IsGate` over `HasGate` because it's not about rooms, rather it's about `RoomType`, the `RoomType` can be "Gate", but there's no way to define whether the `RoomType` HAS the specified gate, without manually doing some checks, which wouldn't lead to extension(s) method(s) anymore; that's an API-related feature.
Let's just keep it as it as at the moment.